lambdas/service/app.py

```diff
@@ -655,204 +655,270 @@ def repository_search(entity_type: str, item_id: str):
         raise NotFoundError(msg=e)
 
 
-@app.route('/repository/files', methods=['HEAD', 'GET'], cors=True)
-@app.route('/repository/files/{file_id}', methods=['GET'], cors=True)
+generic_object_spec = schema.object(additional_properties=True)
+array_of_object_spec = schema.array(generic_object_spec)
+hit_spec = schema.object(
+    additional_properties=True,
+    protocols=array_of_object_spec,
+    entryId=str,
+    samples=array_of_object_spec,
+    specimens=array_of_object_spec,
+    cellLines=array_of_object_spec,
+    donorOrganisms=array_of_object_spec,
+    organoids=schema.array(str),
+    cellSuspensions=array_of_object_spec,
+)
+
+page_spec = schema.object(
+    hits=schema.array(hit_spec),
+    pagination=generic_object_spec,
+    termFacets=generic_object_spec
+)
+
+
+def filters_param_spec(facets):
+    array_pair_spec = schema.array({}, minItems=2, maxItems=2)
+    return params.query(
+        'filters',
+        schema.optional(
+            application_json(
+                # Swagger UI doesn't produce a useful example for this schema,
+                # so several are manually provided below.
+                schema.object(**{
+                    facet: schema.optional({
+                        'oneOf': [
+                            schema.object(**{'is': schema.array({})}),
+                            *[
+                                schema.object(**{rel: array_pair_spec})
+                                for rel in ['contains', 'within', 'intersects']
+                            ]
+                        ]
+                    })
+                    for facet in facets
+                })
+            )
+        ),
+        description=format_description('''
+            Criteria to filter entities from the search results.
+
+            Each filter consists of a facet, a relationship, and an array of
```

Comment:
```suggestion
            Each filter consists of a facet, a relational operator, and an array of
```

here and then just `operator` henceforth.

Reply:
```suggestion
            Each filter consists of a facet name, a relationship, and an array of
```